java/src/main/native/CMakeLists.txt

```diff
@@ -254,8 +256,15 @@ target_include_directories(
 # Override RPATH for cudfjni
 set_target_properties(
   cudfjni
-  PROPERTIES BUILD_RPATH "\$ORIGIN" INSTALL_RPATH "\$ORIGIN" # set target compile options
-             CXX_STANDARD 17 CXX_STANDARD_REQUIRED ON CUDA_STANDARD 17 CUDA_STANDARD_REQUIRED ON
+  PROPERTIES BUILD_RPATH "\$ORIGIN"
+             INSTALL_RPATH "\$ORIGIN"
+             # set target compile options
+             CXX_STANDARD 17
+             CXX_STANDARD_REQUIRED ON
+             CUDA_STANDARD 17
+             CUDA_STANDARD_REQUIRED ON
+             POSITION_INDEPENDENT_CODE ON
```

Comment:
I think I understand why this change was required, but I'd better confirm.

Before this PR, `-fPIC` was implied already because `libcudfjni.so` was always `SHARED`. But now that we might be building archives, `-fPIC` is no longer implied, as per [CMake docs](https://cmake.org/cmake/help/latest/prop_tgt/POSITION_INDEPENDENT_CODE.html). So this needs to be added explicitly?

Reply:
Yes, if we don't specify this explicitly then when building a static library instead of shared it will not compile PIC by default.  However we need the code compiled PIC in order to use it in another shared library of an external project.